scripts/Connector/Connector_JST/conn_jst_XA_horizontal.py

```diff
@@ -0,0 +1,333 @@
+                ])
+            polygons.append([
+                {'x': x_min + side, 'y': silk_y_max - 2.3},
+                {'x': x_min + side, 'y': 2},
+                {'x': silk_x_min, 'y': 2},
+                ])
+        for poly in polygons:
+            kicad_mod.append(PolygoneLine(polygone = poly,
+                layer='F.SilkS', width=configuration['silk_line_width']))
+
+    ########################### CrtYd ################################
+    part_x_min = x_min
+    part_x_max = x_max
+    part_y_min = y_min
+    part_y_max = y_max
+    offset = configuration['courtyard_offset']['connector']
+    grid = configuration['courtyard_grid']
+
+    cx1 = roundToBase(part_x_min - offset, grid)
+    cy1 = roundToBase(part_y_min - offset, grid)
+
+    cx2 = roundToBase(part_x_max + offset, grid)
+    cy2 = roundToBase(part_y_max + offset, grid)
+
+    cx3 = roundToBase(part_x_max - body_back_protrusion_width - offset, grid)
+    cy3 = roundToBase(-pad_size[1]/2 - offset, grid)
+
+    cx4 = roundToBase(part_x_min + body_back_protrusion_width + offset, grid)
+
+    kicad_mod.append(PolygoneLine(polygone=[
+            {'x':cx1, 'y':cy1},
+            {'x':cx1, 'y':cy2},
+            {'x':cx2, 'y':cy2},
+            {'x':cx2, 'y':cy1},
+            {'x':cx3, 'y':cy1},
+            {'x':cx3, 'y':cy3},
+            {'x':cx4, 'y':cy3},
+            {'x':cx4, 'y':cy1},
+            {'x':cx1, 'y':cy1},
+            ],
+        layer='F.CrtYd', width=configuration['courtyard_line_width']))
+
+    ########################### Fab Outline ################################
+    tmp_x1=x_min+body_back_protrusion_width
+    tmp_x2=x_max-body_back_protrusion_width
+    poly_fab_outline= [
+                    {'x':tmp_x1, 'y':y_main_min},
+                    {'x':tmp_x1, 'y':y_min},
+                    {'x':x_min, 'y':y_min},
+                    {'x':x_min, 'y':y_max},
+                    {'x':x_max, 'y':y_max},
+                    {'x':x_max, 'y':y_min},
+                    {'x':tmp_x2, 'y':y_min},
+                    {'x':tmp_x2, 'y':y_main_min},
+                    {'x':tmp_x1, 'y':y_main_min}
+    ]
+    kicad_mod.append(PolygoneLine(polygone=poly_fab_outline, layer='F.Fab', width=configuration['fab_line_width']))
+
+    ############################# Pads ##################################
+    optional_pad_params = {}
+    if configuration['kicad4_compatible']:
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_RECT
+    else:
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_ROUNDRECT
+
+    kicad_mod.append(PadArray(initial=1, start=[0, 0],
+        x_spacing=pitch, pincount=pincount,
+        size=pad_size, drill=drill_size,
+        type=Pad.TYPE_THT, shape=Pad.SHAPE_OVAL, layers=Pad.LAYERS_THT,
+        **optional_pad_params))
+
+    #add mounting hole for hook
+    if hook:
+        kicad_mod.append(Pad(at={'x': A/2, 'y': 6.5}, type=Pad.TYPE_NPTH, shape=Pad.SHAPE_OVAL, layers=Pad.LAYERS_NPTH,
```

Comment:
For the y position, please calculate from the datasheet values (9.2 - 2.7) instead of a magic number.